syntax = "proto3";
package management;
import "google/protobuf/empty.proto";

message StartRequest {
  string bot_id = 1;
}

message BotIdRequest {
  string bot_id = 1;
}

message StatusResponse {
  string status = 1;
}

message BotStatusRequest {
  string bot_id = 1;
  int32 lines = 2;
}

message BotStatusResponse {
  string bot_id = 1;
  bool running = 2;
  int32 pid = 3;
  int32 returncode = 4;
  string logs = 5;
}

message LogRequest {
  int32 lines = 1;
}

message LogResponse {
  string logs = 1;
}

message BotList {
  map<string, bool> bots = 1;
}

message ConfigChange {
  string key = 1;
  string value = 2;
  string reason = 3;
}

service ManagementService {
  rpc StartBot(StartRequest) returns (StatusResponse);
  rpc StopBot(BotIdRequest) returns (StatusResponse);
  rpc BotStatus(BotStatusRequest) returns (BotStatusResponse);
  rpc GetLogs(LogRequest) returns (LogResponse);
  rpc ListBots(google.protobuf.Empty) returns (BotList);
  rpc UpdateConfig(ConfigChange) returns (StatusResponse);
}
